example/scalalib/native/4-common-config/build.mill

```diff
@@ -0,0 +1,121 @@
+// This example shows some of the common tasks you may want to override on a
+// `{language}Module`: specifying the `mainClass`, adding additional
+// sources/resources, generating resources, and setting compilation/run
+// options.
+
+//// SNIPPET:BUILD
+package build
+import mill._, scalalib._, scalanativelib._
+
+object `package` extends RootModule with ScalaNativeModule {
+  def scalaVersion = "2.13.11"
+  def scalaNativeVersion = "0.5.5"
+
+  // You can have arbitrary numbers of third-party dependencies
+  def ivyDeps = Agg(
+    ivy"com.lihaoyi::fansi::0.5.0",
+    ivy"com.lihaoyi::os-lib::0.11.1"
+  )
+
+  // Choose a main class to use for `.run` if there are multiple present
+  def mainClass: T[Option[String]] = Some("foo.Foo2")
+  
+  // Add (or replace) source folders for the module to use
+  def sources = T.sources{
+    super.sources() ++ Seq(PathRef(millSourcePath / "custom-src"))
+  }
+
+  // Add (or replace) resource folders for the module to use
+  def resources = T.sources{
+    super.resources() ++ Seq(PathRef(millSourcePath / "custom-resources"))
+  }
+
+  // Generate sources at build time
+  def generatedSources: T[Seq[PathRef]] = T {
+    for(name <- Seq("A", "B", "C")) os.write(
+      T.dest / s"Foo$name.scala",
+      s"""
+         |package foo
+         |object Foo$name {
+         |  val value = "hello $name"
+         |}
+      """.stripMargin
+    )
+
+    Seq(PathRef(T.dest))
+  }
```

Comment:
We don't need all these standard Scala config options here. Only Scala-Native specific ones